syntax="proto3";

package com.demo.app;

message AccountDetails{
  int32 accountnumber=1;
  string name=2;
  float balance=3;
}

message AccountRequest{
  int32 accountnumber=1;
}

message CreateAccountRequest{
  string name=1;
  float initial_balance=2;
}
service AccountService{
  rpc GetAccountDetails(AccountRequest) returns (AccountDetails);
  rpc CreateAccount(CreateAccountRequest) returns (AccountDetails);
}
